catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -0,0 +1,383 @@
+            queryResponse.setHits(HITS_RESULT);
+            QueryRequestImpl queryRequest = (QueryRequestImpl) args[3];
+            assertThat(queryRequest.getPropertyValue(QUERY_MODE), is(UPDATE_QUERY_MODE));
+            return mockSortedQueryMonitor;
+        });
+
+        QueryResponse federateResponse = strategy.federate(sources, fedQueryRequest);
+
+        WaitCondition.expect("Wait until the result appears in the response")
+                .until(() -> federateResponse.getHits(), is(HITS_RESULT))
+                .checkEvery(POLLING_INTERVAL, TimeUnit.MILLISECONDS)
+                .within(WITHIN_TIMEOUT, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testFederateQueryCacheWithOffsetHandler() throws Exception {
+
+        HashMap<String, Serializable> properties = new HashMap<>();
+        properties.put(QUERY_MODE, INDEX_QUERY_MODE);
+
+        // startIndex and pageSize must be > 1
+        Query mockQ = new QueryImpl(mock(NullFilterImpl.class),
+                2,
+                5,
+                mock(SortBy.class),
+                true,
+                1000);
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQ, properties);
+
+        List<Source> sources = new ArrayList<>();
+
+        // Multiple sources needed for OffsetResultHandler to be used
+        for (int i = 0; i < 2; i++) {
+            sources.add(getMockSource(mockResponse));
+        }
+
+        when(mockSortedQueryMonitorFactory.createMonitor(any(),
+                any(),
+                any(),
+                any(),
+                any())).thenAnswer(invocation -> {
+            Object[] args = invocation.getArguments();
+            QueryResponseImpl queryResponse = (QueryResponseImpl) args[2];
+            queryResponse.setHits(HITS_RESULT);
+            QueryRequestImpl queryRequest = (QueryRequestImpl) args[3];
+            assertThat(queryRequest.getPropertyValue(QUERY_MODE), is(INDEX_QUERY_MODE));
+            return mockSortedQueryMonitor;
+        });
+
+        QueryResponse response = strategy.federate(sources, fedQueryRequest);
+
+        // Response is changed by OffsetResultHandler, so getHits() returns 0
+        WaitCondition.expect("Wait until the result appears in the response")
+                .until(() -> response.getHits(), not(42L))
+                .checkEvery(POLLING_INTERVAL, TimeUnit.MILLISECONDS)
+                .within(WITHIN_TIMEOUT, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testUpdateResponse() throws Exception {
+        Map<String, Serializable> testMap = new HashMap<>();
+        testMap.put(Constants.SERVICE_TITLE, MOCK_RESPONSE_TITLE);
+        UpdateResponse response = new UpdateResponseImpl(mock(UpdateRequestImpl.class),
+                testMap,
+                new ArrayList<>());
+
+        response = strategy.process(response);
+
+        assertThat(response.getPropertyValue(Constants.SERVICE_TITLE), is(MOCK_RESPONSE_TITLE));
```

Comment:
❗️ We need to verify that `SolrCache.create()` was called with the expected metacards (L404).